src/main/java/com/microsoft/azure/relay/HybridHttpConnection.java

```diff
@@ -415,12 +422,10 @@ public String toString() {
 					ListenerCommand.ResponseCommand responseCommand = createResponseCommand(this.context);
 					if (this.writeBufferStream != null) {
 						responseCommand.setBody(true);
-						this.writeBufferStream.position(0);
 					}
-
+					
 					// Don't force any rendezvous now
 					sendTask = this.connection.sendResponseAsync(responseCommand, this.writeBufferStream, this.writeTimeout);
-					this.responseCommandSent = true;
```

Comment:
>responseCommandSent [](start = 10, length = 19)

This is now getting set later, is that intentional?

Reply:
Just putting it in the callback after the send completes.

---
In reply to: [276024545](https://github.com/Azure/azure-relay-java/pull/39#discussion_r276024545) [](ancestors = 276024545)

Reply:
A comment that this makes it closer to the C# implementation (if you had compared them yourself) might have saved me some time.

---
In reply to: [276036735](https://github.com/Azure/azure-relay-java/pull/39#discussion_r276036735) [](ancestors = 276036735,276024545)